x/group/internal/math/dec.go

```diff
@@ -0,0 +1,67 @@
+package math
+
+import (
+	"github.com/cockroachdb/apd/v2"
+
+	"github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// Dec is a wrapper struct around apd.Decimal that does no mutation of apd.Decimal's when performing
+// arithmetic, instead creating a new apd.Decimal for every operation ensuring usage is safe.
+//
+// Using apd.Decimal directly can be unsafe because apd operations mutate the underlying Decimal,
+// but when copying the big.Int structure can be shared between Decimal instances causing corruption.
+// This was originally discovered in regen0-network/mainnet#15.
+type Dec struct {
+	dec apd.Decimal
+}
+
+const mathCodespace = "math"
+
+var ErrInvalidDecString = errors.Register(mathCodespace, 1, "invalid decimal string")
```

Comment:
should this not exist in `types/errors.go` since we're registering it?

Reply:
well for now we decided to put this math package as an `internal` package to `x/group` (which is a standalone go module) so that we don't expose it as part of sdk releases so I thought it made sense to register the error within `x/group/internal/math`

Reply:
@blushi should I revert this and register the error within `x/group/internal/math` ?

Reply:
let's wait for @alexanderbez input on this

Reply:
> well for now we decided to put this math package as an `internal` package to `x/group` (which is a standalone go module) so that we don't expose it as part of sdk releases so I thought it made sense to register the error within `x/group/internal/math`

@alexanderbez any comments on this ?

Reply:
Errors are naturally a public thing to consume and examine, so I **do not** think they belong in `internal`. Consider the case where the error is wrapped/bubbled up stream and the consumer wants the check the error type(s).

Reply:
But this introduces a weird dependency between two go modules which we are trying to avoid. They can be public in x/group but I want to do whatever we can to reduce the coupling of everything to types/